### ISSUE TITLE
Uploads get deleted when localizing sibling fields

### ISSUE THREAD

**Issue by ChrisGV04** [payloadcms/payload#2098]
# Bug Report

I have an Uploads collection with only an "Alt text" sibling field that supports localization. If I upload a file and set the Alt text on the default locale everything works fine, however, when I switch to another language and localize the Alt field, the upload gets deleted from the disk but somehow it still displays on the collection page. The upload is not deleted if I don't localize the Alt text field, but as soon as I localize it, it just disappears from my machine.

https://user-images.githubusercontent.com/47041342/218943048-03d3ea97-35ed-484a-ae49-0452261bd6c3.mp4

## Steps to Reproduce

1. Create a basic Payload project that supports content localization. I used the following configs:

```ts
localization: {
    locales: ['en', 'es'],
    defaultLocale: 'en',
    fallback: true,
},
```

2. Create a basic Uploads collection with an `alt` sibling field that supports localization.

```ts
export const Uploads: CollectionConfig = {
  slug: 'uploads',
  access: {
    read: (): boolean => true,
  },
  admin: { useAsTitle: 'alt' },
  upload: {
    staticURL: '/uploads',
    staticDir: '../public/uploads',
    mimeTypes: ['image/*', 'video/*'],
  },
  fields: [
    {
      name: 'alt',
      type: 'text',
      required: true,
      localized: true,
      label: 'Alt text',
    },
  ],
};
```

3. Upload any image and set the `Alt text` field, then hit "Save". The file should now be saved on your disk.
4. Switch localization to another language and write some other text on the Alt field as with the previous step, then hit "Save".
5. The upload still shows on the collection list, but when I view the uploads folder on my disk, the image is gone.

## Other Details

Payload: 1.6.10

**Comment by jacobsfletch** [comment]
@ChrisGV04 this was a bug outside localization too. I've got a fix for this in #2101 that we'll release asap.

**Comment by ChrisGV04** [comment]
@jacobsfletch Wow, thank you very much!
